src/js/stores/PluginStore.js

```diff
@@ -16,7 +16,9 @@ var storeData = {
 };
 
 function addPlugin(data) {
-  if (data.id == null && !Util.isObject(data.info) &&
+  if (data.id == null || !Util.isObject(data.info) ||
+      !Util.isArray(data.info.modules) ||
+      !data.info.modules.includes(PluginModules.UI) ||
```

Comment:
I kinda like the idea of extracting these checks into their own functions, but we don't need them anywhere else than in this particular case, so I don't see much value in it. Let's keep them as they are.